extensions/api/data-management/transferprocess/src/main/java/org/eclipse/dataspaceconnector/api/datamanagement/transferprocess/TransferProcessApiController.java

```diff
@@ -58,63 +73,104 @@ public TransferProcessApiController(Monitor monitor) {
                 .sortField(sortField)
                 .filter(filterExpression)
                 .sortOrder(sortOrder).build();
-        monitor.debug(format("get all TransferProcesses %s", spec));
-
-        return Collections.emptyList();
+        monitor.debug(format("Get all TransferProcesses %s", spec));
 
+        return service.query(spec).stream()
+                .map(tp -> transformerRegistry.transform(tp, TransferProcessDto.class))
+                .filter(Result::succeeded)
+                .map(Result::getContent)
+                .collect(Collectors.toList());
     }
 
     @GET
-    @Path("{id}")
+    @Path("/{id}")
     @Override
     public TransferProcessDto getTransferProcess(@PathParam("id") String id) {
-        monitor.debug(format("get TransferProcess with ID %s", id));
-
-        return null;
+        monitor.debug(format("Get TransferProcess with ID %s", id));
+
+        return Optional.of(id)
+                .map(service::findById)
+                .map(it -> transformerRegistry.transform(it, TransferProcessDto.class))
+                .filter(Result::succeeded)
+                .map(Result::getContent)
+                .orElseThrow(() -> new ObjectNotFoundException(TransferProcess.class, id));
     }
 
     @GET
-    @Path("{id}/state")
+    @Path("/{id}/state")
     @Override
     public String getTransferProcessState(@PathParam("id") String id) {
-        monitor.debug(format("get TransferProcess State with ID %s", id));
+        monitor.debug(format("Get TransferProcess State with ID %s", id));
 
-        return "";
+        return Optional.of(id)
+                .map(service::getState)
+                .orElseThrow(() -> new ObjectNotFoundException(TransferProcess.class, id));
     }
 
     @POST
+    @Path("/request")
```

Comment:
Seems like these changes will also required us to update openapi specs e.g. resources/openapi/yaml/transferprocess.yaml

To try out I used exact same commands from `.github/workflows/generate_openapi_spec.yaml` and it did show changes for transferprocess controller but it also shows some of the changes for other open api specs as well, so It feels like in EDC we do not have an automated check which verifies if opneapi spec are up to date or not.

Reply:
The gradle check you mentioned seems to just generate the open api specs but doesn't commit it nor checks if the specs are different. Perhaps it would make sense to create issue to modify the check to fail if there are changes detected. 
As for your comment - I updated the openapi specs

Reply:
Yes, agree. OpenAPI check should report a failure in pipeline if generated files are not as expected.